### ISSUE TITLE
Support for Nautilus 43.0 and GTK 4

### ISSUE THREAD

**Issue by samuel-cavalcanti** [talklittle/nautilus-extension-rs#11]
Hi, is there any intention to support new nautilus versions  ?

**Comment by talklittle** [comment]
Yes I would like for there to be support for GNOME 42 and GTK 4, but I am not actively looking into this at the moment.

**Comment by samuel-cavalcanti** [comment]
How hard is to create this rust wrapper ? Maybe I could help you

**Comment by talklittle** [comment]
The first step would be finding the relevant documentation for libnautilus-extension. Migration documents, the updated C header files and API documentation.

Then see how the Nautilus extensions API has changed. If it's similar enough with backward compatibility, we can patch it into the current wrapper. Otherwise we might need to introduce a feature flag to tell which Nautilus API (3 or 4) we are targeting at build time. Then we might end up with two similar copies of the wrapper code.

**Comment by samuel-cavalcanti** [comment]
some documentations: 
- the libnautilus-extension docs  https://archlinux.org/packages/extra/x86_64/libnautilus-extension-docs/
-  nautlilus-python Migration document  to Nautilus API 4.0: https://gnome.pages.gitlab.gnome.org/nautilus-python/nautilus-python-migrating-to-4.html

maybe porting new nautilus API with rust bindgen: https://github.com/rust-lang/rust-bindgen
tutorial videos: [Crust of Rust: Build Scripts and Foreign-Function Interfaces (FFI)]( https://www.youtube.com/watch?v=pePqWoTnSmQ&ab_channel=JonGjengset)
maybe some refactoring, like moving struct NautilusFileInfo to  nautilus_file_info.rs to make easier to compare with C headers files
